assessment.md

```diff
@@ -143,39 +143,96 @@ TODO: add description for each assessment level for this skill.
+
+* None = The program demonstrates no understanding or implementation of software architecture.
+* Inadequate = The program displays experience with software architecture, but there are flow and design issues.
+* Adequate = The program displays experience and understanding of software architecture with minimal flow and design issues.
+* Good = the program displays an extensive knowledge of software architecture theory and underlying experience with no flow or design issues.
+* Excellent = the program displays an extensive knowledge of software theory and underlying experience as a means of conducting and managing the design/flow. Develops concepts and is able to help other team members with the architecture of their software.
 
 * Software design
 
-TODO: add description for each assessment level for this skill.
+Creating the best software blueprint to most efficiently solve a particular problem.
+
+* None = Doesn't even attempt to solve the problem.
+* Inadequate = Creates incorrect software to solve the problem.
+* Adequate = Creates correct software to solve the problem.
+* Good = Creates correct and efficient software to solve the problem.
+* Excellent = Creates correct and efficient software while constantly thinking of how to make it more efficient.
 
 * Software documentation
 
-TODO: add description for each assessment level for this skill.
+Software Documentation will be evaluated on this scale
+
+* None = No explanation of what the software does, no solution to the problem presented, and no example of where the code be used is included
+* Inadequate = A short explanation, no solution to the problem, and no example is included
+* Adequate = A good explanation, a short solution to the problem, and a short example is included
+* Good = A detailed explanation, a solution to the problem, a short example is included
+* Excellent = A detailed explanation, a detailed solution to the problem, a interactive example is included
 
 * Programming styles
 
-TODO: add description for each assessment level for this skill.
+Programming styles will be evaluated on the following levels:
+
+* None = Does not have the ability to program at all in any style.
+* Inadequate = Programming style is completely akin to "spaghetti code."
+* Adequate = Programming style is slightly akin to "spaghetti code." with some correct naming conventions, indentations, and little or non-descriptive commenting.
+* Good = Programming style is fairly thorough and mostly uses proper naming conventions, indentations, and a fair amount of commenting. Styling is mostly consistent at this level.
+* Excellent = Programming style is very thorough and uses proper naming conventions, indentations, and is descriptively commented throughout. Styling is consistent at this level.
 
 * Managing software complexity
 
-TODO: add description for each assessment level for this skill.
+Managing software complexity will be evaluated on the following scale:
+
```

Comment:
These two lines can be removed for consistency with other sections of the document